### DISCUSSION TITLE
How can I open a .insomnia folder?

### DISCUSSION THREAD

**Discussion by robinmollah** [Kong/insomnia#4972] [Help]
I can import and export files. But I have pushed some content to Github. Now, I don't want them in the cloud repository. I have copied them into the local file system. Now, how can I import the contents of the `.insomnia` folder contents?

**Comment by eddex** [comment]
1. Open the design document hosted on Github (the git repo)
2. Click settings and export the document
3. Create a new document
4. Import the file

**Reply by robinmollah** [reply]
I want to keep the `.insomnia` files in a folder where the source code is located. And I want to version control both the source code and the insomnia design document. So that, if I add an API, I can also add the insomnia document changes in the same commit.

**Reply by robinmollah** [reply]
Thanks for your answer. I appreciate it.

**Reply by eddex** [reply]
Ah I see your problem. I actually had the same. I don't think this is possible.

Insomnia clones the git repo into your app data folder. E.g.: `C:\Users\eddex\AppData\Roaming\Insomnia\version-control\git\`.

But it doesn't simply clone the repo, it does some more stuff to the files so you can't use the files in that location like a normal git repo.

What you can do is to create a feature branch for your code changes in your normal repo, then create a branch for your insomnia changes in insomnia. Then you can push your insomnia branch and merge your insomnia branch into your feature branch. If you want you could then proceed to squash the commits in your feature branch. This way only 1 commit will be added to the main branch with all your code and insomnia changes.

**Reply by faustool** [reply]
This is pretty bad... There are some bugs in Insomnia that I could overcome simply by manually changing the file, but I can't do that.
